drivers/ieee802154/ieee802154_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,735 @@
+#include <driverlib/prcm.h>
+#include <driverlib/rf_common_cmd.h>
+#include <driverlib/rf_data_entry.h>
+#include <driverlib/rf_ieee_cmd.h>
+#include <driverlib/rf_mailbox.h>
+#include <driverlib/rfc.h>
+#include <inc/hw_fcfg1.h>
+
+#ifdef CONFIG_SOC_SERIES_CC13X2_CC26X2
+#include <rf_patches/rf_patch_cpe_ieee_802_15_4.h>
+#endif /* CONFIG_SOC_SERIES_CC13X2_CC26X2 */
+
+/* IEEE 802.15.4-2006 2400–2483.5 MHz O-QPSK PHY (6.5.1) */
+#define IEEE802154_OQPSK_SYMBOLS_PER_SECOND 625000
+
+/* IEEE 802.15.4-2006 PHY constants (6.4.1) */
+#define IEEE802154_MAX_PHY_PACKET_SIZE 127
+#define IEEE802154_TURNAROUND_TIME 12
+
+/* IEEE 802.15.4-2006 PHY PIB attributes (6.4.2) */
+#define IEEE802154_PHY_CCA_MODE 1
+#define IEEE802154_PHY_SHR_DURATION 10
+#define IEEE802154_PHY_SYMBOLS_PER_OCTET 2
+
+/* IEEE 802.15.4-2006 MAC constants (7.4.1) */
+#define IEEE802154_UNIT_BACKOFF_PERIOD 20
+
+#define IEEE802154_ACK_PACKET_SIZE 6
+
+/* IEEE 802.15.4-2006 PHY PIB attributes (7.4.2 and 7.5.6.4.2)
+ *
+ * The MAC ACK wait duration is the time commencing transmitting the ACK plus
+ * the length of the ACK frame for a non-beacon enabled PAN.
+ */
+#define IEEE802154_MAC_ACK_WAIT_DURATION                                       \
+	(IEEE802154_TURNAROUND_TIME + IEEE802154_PHY_SHR_DURATION +            \
+	 IEEE802154_ACK_PACKET_SIZE * IEEE802154_PHY_SYMBOLS_PER_OCTET)
+
+#define IEEE802154_MTU (IEEE802154_MAX_PHY_PACKET_SIZE - 2)
+#define OPENTHREAD_MTU 1280
+
+#define CC13XX_CC26XX_RAT_CYCLES_PER_SECOND 4000000
+
+#define CC13XX_CC26XX_NUM_RX_BUF 2
+#define CC13XX_CC26XX_RX_BUF_SIZE (IEEE802154_MAX_PHY_PACKET_SIZE + 2)
+
+#define CC13XX_CC26XX_CPE0_IRQ 9
+#define CC13XX_CC26XX_CPE1_IRQ 2
+
+/* Overrides from SmartRF Studio 7 2.13.0 */
+static u32_t overrides[] = {
+#ifdef CONFIG_SOC_SERIES_CC13X2_CC26X2
+	/* DC/DC regulator: In Tx, use DCDCCTL5[3:0]=0x3 (DITHER_EN=0 and IPEAK=3). */
+	0x00F388D3,
+	/* Rx: Set LNA bias current offset to +15 to saturate trim to max (default: 0) */
+	0x000F8883,
+#endif /* CONFIG_SOC_SERIES_CC13X2_CC26X2 */
+	0xFFFFFFFF
+};
+
+DEVICE_DECLARE(ieee802154_cc13xx_cc26xx);
+
+struct ieee802154_cc13xx_cc26xx_data {
+	struct net_if *iface;
+
+	u8_t ext_addr[8];
+	u16_t short_addr;
+	u16_t pan_id;
+
+	struct k_sem tx_done;
+	struct k_sem rx_done;
+	struct k_sem ack;
+
+	K_THREAD_STACK_MEMBER(rx_stack,
+			      CONFIG_IEEE802154_CC13XX_CC26XX_RX_STACK_SIZE);
+	struct k_thread rx_thread;
+
+	dataQueue_t rx_queue;
+	rfc_dataEntryPointer_t rx_entry[CC13XX_CC26XX_RX_BUF_SIZE];
+	u8_t rx_data[CC13XX_CC26XX_NUM_RX_BUF]
+		    [CC13XX_CC26XX_RX_BUF_SIZE] __aligned(4);
+
+	volatile rfc_CMD_IEEE_CCA_REQ_t cmd_ieee_cca_req;
+	volatile rfc_CMD_CLEAR_RX_t cmd_clear_rx;
+	volatile rfc_CMD_IEEE_RX_t cmd_ieee_rx;
+	volatile rfc_CMD_SET_TX_POWER_t cmd_set_tx_power;
+	volatile rfc_CMD_IEEE_CSMA_t cmd_ieee_csma;
+	volatile rfc_CMD_IEEE_TX_t cmd_ieee_tx;
+	volatile rfc_CMD_IEEE_RX_ACK_t cmd_ieee_rx_ack;
+	volatile rfc_CMD_RADIO_SETUP_t cmd_radio_setup;
+};
+
```

Comment:
I'd rather see all the above in a private local header

Reply:
Added a private header.